flash/core/classification.py

```diff
@@ -22,14 +22,22 @@
 
 class ClassificationPostprocess(Postprocess):
 
+    def __init__(self, multi_label: bool = False, save_path: Optional[str] = None):
+        super().__init__(save_path=save_path)
+        self.multi_label = multi_label
+
     def per_sample_transform(self, samples: Any) -> Any:
-        return torch.argmax(samples, -1).tolist()
+        if self.multi_label:
+            return F.sigmoid(samples).tolist()
+        else:
+            return torch.argmax(samples, -1).tolist()
 
 
 class ClassificationTask(Task):
 
-    def __init__(self, *args, **kwargs) -> None:
-        super().__init__(*args, default_postprocess=ClassificationPostprocess(), **kwargs)
+    postprocess_cls = ClassificationPostprocess
 
     def to_metrics_format(self, x: torch.Tensor) -> torch.Tensor:
+        if getattr(self.hparams, "multi_label", False):
+            return F.sigmoid(x).int()
```

Comment:
Some metrics (e.g. AUROC) require just probabilities, maybe casting to int is a bad idea?

Reply:
Yes, removed !